MobiFlight/Joysticks/Joystick.cs

```diff
@@ -216,32 +246,33 @@ public List<ListItem<IBaseDevice>> GetAvailableDevicesAsListItems()
 
         private List<JoystickDevice> GetButtonsSorted()
         {
-            var buttons = Buttons.ToArray().ToList();
-            buttons.Sort(SortByPositionInDefintion);
-            return Buttons;
+            return Buttons.OrderBy(button => GetIndexForJoystickDevice(button)).ToList();
         }
 
         private List<JoystickDevice> GetAxisSorted()
         {
-            var axes = Axes.ToArray().ToList();
-            Axes.Sort(SortByPositionInDefintion);
-
-            return Axes;
+            return Axes.OrderBy(axis => GetIndexForJoystickDevice(axis)).ToList();
         }
 
-        public int GetIndexForKey(string key)
+        public virtual int GetIndexForKey(string key)
         {
-            return Definition?.Inputs?.FindIndex(input => input.Name == key) ?? 0;
+            // -2 = definition not found / no inputs in definition
+            // -1 = index not found
+            // 0 or greater: valid index
+            return Definition?.Inputs?.FindIndex(input => input.Name == key) ?? -2;
         }
 
-        int SortByPositionInDefintion(JoystickDevice b1, JoystickDevice b2)
+        public int GetIndexForJoystickDevice(JoystickDevice button)
```

Comment:
Enumerable.OrderBy, unlike List.Sort, allows us to define what property to sort by. Therefore, the approach to integrate the index retrieval into the function is somewhat different.